lib/common/socket.c

```diff
@@ -999,6 +1039,39 @@ void h2o_socket_ssl_async_resumption_setup_ctx(SSL_CTX *ctx)
     /* if necessary, it is the responsibility of the caller to disable the internal cache */
 }
 
+static void on_dispose_ssl_ctx_ex_data(void *parent, void *ptr, CRYPTO_EX_DATA *ad, int idx, long argl, void *argp)
+{
+    h2o_cache_t *ssl_session_cache = (h2o_cache_t *)ptr;
+    if (ssl_session_cache != NULL)
+        h2o_cache_destroy(ssl_session_cache);
+}
+
+static int get_ssl_session_cache_index()
```

Comment:
In C, you should always specify `(void)` when declaring a function that takes zero arguments (notice that this line works as both declaration and definition).

You may interested in reading http://d.hatena.ne.jp/eel3/20141005/1412521223 to understand the reason behind the necessity.